internal/common/environment/IEnvironmentObject.go

```diff
@@ -0,0 +1,19 @@
+package ienvironmentobject
+
+/*
+
+The IEnvironementObject is an interface class that all objects (including agents) must implement.
+
+*/
+
+import (
+	"github.com/google/uuid"
+)
+
+type IEnvironementObject[T any] struct {
+	// returns the unique ID of an object
+	GetID() uuid.UUID
+
+	// returns the current coordinates of Agent
+	GetPosition() [2]float64
```

Comment:
I'd add a `commonTypes.go` in this package to redefine the position as its own data structure

`type Coordinate struct {
  x float32
y float32
}
`